### PR TITLE
fix(): wait for gw pods to be in ready state before marking complete

### PR DESCRIPTION
**Pull request by YachikaRalhan**

### CODE REVIEW

File: pkg/hub/controllers/vpnkeyrotation/reconciler.go [hunk 1]
```diff
@@ -380,9 +380,15 @@ func (r *Reconciler) syncCurrentRotationState(ctx context.Context,
 		syncedRotationState := make(map[string]hubv1alpha1.StatusOfKeyRotation)
 		for _, gw := range allGwsUnderCluster {
 			obj, ok := currentRotationState[gw]
+
 			if ok {
 				syncedRotationState[gw] = obj
 			} else if !ok {
+				gwIsReady := r.areGatewayPodsReady(ctx, gw)
+				if !gwIsReady {
```
Comment (rahulii) [inline pkg/hub/controllers/vpnkeyrotation/reconciler.go:388]: lets just use
if !r.areGatewayPodsReady(ctx, gw)

**Comment by rahulii** [comment]
Lets add UTs and ITs if posibble.

File: pkg/hub/controllers/vpnkeyrotation/reconciler.go [hunk 2]
```diff
@@ -366,10 +366,10 @@ func (r *Reconciler) syncCurrentRotationState(ctx context.Context,
 			return getErr
```
Comment (rajendra-avesha) [inline pkg/hub/controllers/vpnkeyrotation/reconciler.go:366]: @YachikaRalhan this method returns (bool, error) you are returning one value

File: pkg/hub/controllers/vpnkeyrotation/reconciler.go [hunk 3]
```diff
@@ -380,9 +380,15 @@ func (r *Reconciler) syncCurrentRotationState(ctx context.Context,
...
 		for _, gw := range allGwsUnderCluster {
 			obj, ok := currentRotationState[gw]
+
 			if ok {
 				syncedRotationState[gw] = obj
 			} else if !ok {
+				gwIsReady := r.areGatewayPodsReady(ctx, gw)
+				if !gwIsReady {
+					requeue = true
+					return errors.New("gateway pods are not ready")
```
Comment (rajendra-avesha) [inline pkg/hub/controllers/vpnkeyrotation/reconciler.go:390]: @YachikaRalhan you need to requeue also here (bool, error)

File: pkg/hub/controllers/vpnkeyrotation/reconciler.go [hunk 4]
```diff
@@ -474,6 +480,25 @@ func (r *Reconciler) updateCertificates(ctx context.Context, rotationVersion int
...
+		"kubeslice.io/slice-gw": sliceGWName,
+	}
+	listOptions := []client.ListOption{
+		client.MatchingLabels(labels),
+	}
+	err := r.WorkerClient.List(ctx, &podsList, listOptions...)
+	if err != nil {
+		return false
+	}
+	if len(podsList.Items) == 2 {
```
Comment (rajendra-avesha) [inline pkg/hub/controllers/vpnkeyrotation/reconciler.go:496]: @YachikaRalhan will the pod count be always 2?

**Review by rajendra-avesha — CHANGES_REQUESTED** [review]

File: pkg/hub/controllers/vpnkeyrotation/reconciler.go (hunk 3 above)
Comment (YachikaRalhan) [reply pkg/hub/controllers/vpnkeyrotation/reconciler.go:390]: This is under errors.Retry function which is returning only bool.

File: pkg/hub/controllers/vpnkeyrotation/reconciler.go (hunk 4 above)
Comment (YachikaRalhan) [reply pkg/hub/controllers/vpnkeyrotation/reconciler.go:496]: Yes each gateway have two deployments/pods

File: pkg/hub/controllers/vpnkeyrotation/reconciler.go (hunk 2 above)
Comment (YachikaRalhan) [reply pkg/hub/controllers/vpnkeyrotation/reconciler.go:366]: @rajendra-avesha that is under errors.Retry which has a callback function returning error only

**Review by rajendra-avesha — APPROVED** [review]

**Pull request merged**